src/components/events/EventCard.jsx

```diff
@@ -1,24 +1,54 @@
-import { Link } from "react-router-dom";
+import { useSelector } from "react-redux";
+import { selectToken } from "../auth/authSlice";
+import { useUpdateEventMutation } from "./eventsSlice";
+import { useState, useEffect } from "react";
 
-//Tooken import and selector
-// import { useSelector } from "react-redux";
-// import { selectToken } from "../../store/authSlice";
+import { Link } from "react-router-dom";
+import { useGetMeQuery } from "../users/userSlice";
 
 export default function EventCard({ event }) {
   const date = event.startTime.slice(0, 10);
   const time = event.startTime.slice(11, 19);
 
-  // const token = useSelector(selectToken);
+  const token = useSelector(selectToken);
+
+  const { data: user, isLoading, error } = useGetMeQuery();
+  console.log(user);
+
+  //state for changing button classnames when going or not going
+  const [attending, setAttending] = useState(false);
 
-  const token = true;
-  const attending = true;
+  /*Using use effect to only run the logic when the user changes or the attending users
+   Change to run the code only when its needed*/
+  useEffect(() => {
```

Comment:
I don't think the useEffect code here is necessary - the toggleAttendance function will update the attending state already!

Reply:
The useEffect function is to set the initial value of the attending or not in state. That so that we you just open or refresh the page, the buttons wont be defaulted to false or true, it will look int he users events and if the user is there, the going button will be highlighted... also if that don't use useEffect to set the initial value, it will say that the code is running infinitely  and it breaks.

Reply:
Ah, I see. I just caught up with the logic here, thanks for the explanation!

Reply:
No problema, happy  to clarify.